docker/backup.mssql.plugin

```diff
@@ -196,15 +202,29 @@ function onGetDbSize(){
     _hostname=$(getHostname ${flags} ${_databaseSpec})
     _database=$(getDatabaseName ${_databaseSpec})
     _port=$(getPort ${flags} ${_databaseSpec})
-    _portArg=${_port:+"--port ${_port}"}
-    _username=$(getUsername ${_databaseSpec})
+    _portArg=${_port:+",${_port}"}
+    _username=$(getLocalOrDBUsername ${_databaseSpec})
     _password=$(getPassword ${_databaseSpec})
 
-    size=$(sqlcmd -S ${_hostname} -U ${_username} -P ${_password} -d ${_database} -Q "SELECT CONVERT(VARCHAR,SUM(size)*8/1024)+' MB' AS 'size' FROM sys.master_files m INNER JOIN sys.databases d ON d.database_id = m.database_id WHERE d.name = '${_database}' AND m.type_desc = 'ROWS' GROUP BY d.name")
+    size=$(sqlcmd -S ${_hostname}${_portArg} -U ${_username} -P ${_password} -d ${_database} -Q "SELECT CONVERT(VARCHAR,SUM(size)*8/1024)+' MB' AS 'size' FROM sys.master_files m INNER JOIN sys.databases d ON d.database_id = m.database_id WHERE d.name = '${_database}' AND m.type_desc = 'ROWS' GROUP BY d.name")
     rtnCd=${?}
 
     echo ${size}
     return ${rtnCd}
   )
 }
+
+function getLocalOrDBUsername(){
+  (
+    _databaseSpec=${1}
+    _localhost="127.0.0.1"
+    _hostname=$(getHostname ${flags} ${_databaseSpec})
```

Comment:
The value of `${flags}` is implied.  Please ensure the flags are passed explicitly in all calls; for example `getLocalOrDBUsername ${flags} ${_databaseSpec}`.  Or, pass the `hostName` (which has already been resolved before the call to `getLocalOrDBUsername`) and the `databaseSpec` to avoid the additional call to `getHostname`; eliminating the need for the flags.